File: api/src/EloBaza.Application/Queries/Common/PagingInfo.cs
using System;

namespace EloBaza.Application.Queries.Common
{
    /// <summary>
    /// Information about queried resource pagination
    /// </summary>
    public class PagingInfo
    {
        /// <summary>
        /// Total number of records
        /// </summary>
        public int TotalCount { get; private set; }
        /// <summary>
        /// Current page number
        /// </summary>
        public int Page { get; private set; }
        /// <summary>
        /// Current page size
        /// </summary>
        public int PageSize { get; private set; }
        /// <summary>
        /// Last page number
        /// </summary>
        public int LastPage { get; private set; }
        public bool HasNext => Page < LastPage;
        public bool HasPrevious => Page > 1;

        public PagingInfo(int totalCount, int page, int pageSize)
        {
            TotalCount = totalCount;
            Page = page;
            PageSize = pageSize;
            LastPage = (int)Math.Ceiling(totalCount / (double)pageSize);
        }
    }
}

File: ui/src/app/core/models/paging-info.ts
export class PagingInfo {
  totalCount: number;
  page: number;
  pageSize: number;
  lastPage: number;
  get hesPrevious(): boolean {
    return this.page > 1;
  }
  get hasNext(): boolean {
    return this.page < this.lastPage;
  }

  constructor(
    totalCount: number,
    page: number,
    pageSize: number,
    lastPage: number
  ) {
    this.totalCount = totalCount;
    this.page = page;
    this.pageSize = pageSize;
    this.lastPage = lastPage;
  }

  static empty(): PagingInfo {
    return new PagingInfo(0, 0, 0, 0);
  }
}
